pkg/testutil/test_suite_runner.go

```diff
@@ -0,0 +1,167 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package testutil
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	mocksvcsdkapi "github.com/aws-controllers-k8s/elasticache-controller/mocks/aws-sdk-go/elasticache"
+	acktypes "github.com/aws-controllers-k8s/runtime/pkg/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+// TestSuiteRunner runs given test suite config with the help of delegate supplied to it
+type TestSuiteRunner struct {
+	TestSuite *TestSuite
+	Delegate  TestRunnerDelegate
+}
+
+// fixtureContext is runtime context for test scenario given fixture.
+type fixtureContext struct {
+	desired         acktypes.AWSResource
+	latest          acktypes.AWSResource
+	mocksdkapi      *mocksvcsdkapi.ElastiCacheAPI
+	resourceManager acktypes.AWSResourceManager
+}
+
+// expectContext is runtime context for test scenario expectation fixture.
+type expectContext struct {
+	latest acktypes.AWSResource
+	err    error
+}
+
+// TestRunnerDelegate provides interface for custom resource tests to implement.
+// TestSuiteRunner depends on it to run tests for custom resource.
+type TestRunnerDelegate interface {
+	ResourceDescriptor() acktypes.AWSResourceDescriptor
+	Equal(desired acktypes.AWSResource, latest acktypes.AWSResource) bool // remove it when ResourceDescriptor.Delta() is available
+	ResourceManager(*mocksvcsdkapi.ElastiCacheAPI) acktypes.AWSResourceManager
```

Comment:
same here for the generic framework.

Reply:
The generic framework can be moved to ACK `runtime` and the `code-generator` can be updated to generate Service API specific files.
It can be done later in later PRs.